src/client/datascience/notebook/kernelWithMetadata.ts

```diff
@@ -99,9 +99,16 @@ export class VSCodeNotebookKernelMetadata implements VSCNotebookKernel {
         handlerDisposables.push({ dispose: () => kernelDisposedDisposable?.dispose() });
     }
 
-    private chainExecution(next: () => Promise<void>): Promise<void> {
-        const prev = this.pendingExecution ?? Promise.resolve();
-        this.pendingExecution = prev.then(next);
+    private async chainExecution(next: () => Promise<void>): Promise<void> {
+        if (this.pendingExecution) {
+            try {
+                await this.pendingExecution;
+            } catch (e) {
+                // Errors are handled elsewhere
+                traceInfo(`Kernel execution previous failure: ${e}`);
```

Comment:
Is this correct? Doesn't this mean, if you have 5 cells & first cell crashes, then the rest of the 4 cells will execute? that seems to be a change to how things should behave.
I thought in classic jupyter, it stops execution when a cell fails.

Reply:
That's not what is happening here. The pendingExecution will handle the errors itself. This prevents errors from previous runs from not running anything which is what the old code would do.

Reply:
Got it thanks.